firefox_ui_tests/remote/security/test_mixed_script_content_blocking.py

```diff
@@ -0,0 +1,100 @@
+        self.marionette.set_context("chrome")
+        self.marionette.execute_script("arguments[0].hidePopup();",
+                                       script_args=[self.locationbar.notification_popup])
+        FirefoxTestCase.tearDown(self)
+
+    def _expect_protection_status(self, enabled):
+        color, parts, state = (
+            "rgb(255, 0, 0)",
+            "identity-icons-https-mixed-active.png",
+            "unblocked"
+        )
+        if enabled:
+            color, parts, state = (
+                "rgb(0, 136, 0)",
+                "identity-icons-https.png",
+                "blocked"
+            )
+
+        icon_id = "bad-content-%s-notification-icon" % state
+        icon = self.marionette.find_element(By.ID, icon_id)
+        self.wait_for_condition(lambda _: icon.get_attribute('showing') == 'true')
+        favicon = self.locationbar.favicon
+
+        self.wait_for_condition(
+            lambda _: parts in favicon.value_of_css_property('list-style-image')
+        )
+
+        with self.marionette.using_context('content'):
+            for identifier, description in self.test_elements:
+                el = self.marionette.find_element(By.ID, identifier)
+                Wait(self.marionette).until(
+                    lambda mn: el.value_of_css_property('color') == color,
+                    message=("%s has been %s" % (description, state))
+                )
+
+    def expect_protection_enabled(self):
+        self._expect_protection_status(True)
+
+    def expect_protection_disabled(self):
+        self._expect_protection_status(False)
+
+    @skip_under_xvfb
+    def test_mixed_content_page(self):
+        with self.marionette.using_context('content'):
+            self.marionette.navigate(self.test_url)
+
+        self.expect_protection_enabled()
+
+        panel = self.locationbar.notification_popup
+        self.assertEqual(panel.get_attribute('state'), 'closed')
+
+        popupbox = self.marionette.find_element(By.ID,
+                                                'bad-content-blocked-notification-icon')
+        popupbox.click()
+
+        self.wait_for_condition(lambda _: panel.get_attribute('state') == 'open')
+
+        parent_element = self.marionette.find_element(By.ID, 'bad-content-notification')
+
+        button_label = self.browser.get_entity('mixedContentBlocked2.options')
+        options_button = parent_element.find_element(By.ANON_ATTRIBUTE,
+                                                     {'label': button_label})
+        options_button.click()
+
+        item_label = self.browser.get_entity('mixedContentBlocked2.unblock.label')
+        menu_item = parent_element.find_element(By.ANON_ATTRIBUTE,
+                                                {'label': item_label})
+        menu_item.click()
+
+        self.wait_for_condition(lambda _: panel.get_attribute('state') == 'closed')
+
+        self.expect_protection_disabled()
+
+        self.locationbar.reload_url()
```

Comment:
This method does not wait for the page being loaded. I would expect race conditions. Given that this test doesn't test the reload feature of the locationbar we can use `navigate()`.